src/Internal.hh

```diff
@@ -43,9 +43,15 @@ protected:
 };
 
 
+/** CBLStoppable objects can be registered / unregisted to / from the database. The registered object will be called
+    to stop() when the database is closed. The object will be unregistered either after the object is called to stop() or
+    when there is an API call to unregister the object. The database will also retain the object when the object is registered,
+    and will release the object when the object is unregistered to ensure that the object is alive until the object is unregistered. */
 struct CBLStoppable {
     virtual ~CBLStoppable() = default;
-    virtual void stop() = 0;
+    virtual void stopStoppable() = 0;
+    virtual void retainStoppable() = 0;
```

Comment:
Consider inherited from RefCounted instead of this?

Reply:
It doesn't have a simple hierarchy graph.

ListenerToken < CBLQueryChangeListener > 
   <-CBLListenerToken <- CBLRefCounted <- fleece::RefCounted <- fleece::InstanceCountedIn<CBLRefCounted>
   <-CBLStoppable <- CBLRefCounted <- fleece::RefCounted <- fleece::InstanceCountedIn<CBLRefCounted>

ListenerToken < CBLQueryChangeListener > inherits from both  CBLListenerToken and CBLStoppable BUT CBLListenerToken is not CBLStoppable.

Reply:
From the discussion, I have made change by making CBLStoppable wrapping CBLRefCounted and adding CBLQueryListenerStoppable and CBLReplicatorStoppable which extend CBLStoppable and implement the stop() function.